service/matching/matchingEngine.go

```diff
@@ -1216,6 +1204,10 @@ func (e *matchingEngineImpl) getTask(
 		stickyInfo,
 	)
 	if err != nil {
+		if err == errUserDataDisabled {
+			// Rewrite to nicer error message
+			err = serviceerror.NewFailedPrecondition("Operations on versioned workflows are disabled")
```

Comment:
Just noting that the implication here is that this will crash Core based SDKs after a grace period of 1 minute.
I _think_ that's better than silently keeping those workers around idle.

Reply:
Right. We can change this to a NewerBuild error to keep them idle?

Reply:
Not sure, it depends on how transient this state is.
Seems more permanent and less expected than than new build so I tend to think we should crash and be noisy.